tools/github-team-user-store/GitHubTeamUserStore/GitHubTeamUserStore/GitHubEventClient.cs

```diff
@@ -0,0 +1,199 @@
+        /// </summary>
+        /// <param name="prependMessage">Optional message to prepend to the rate limit message.</param>
+        public async Task WriteRateLimits(string prependMessage = null)
+        {
+            var miscRateLimit = await GetRateLimits();
+            CoreRateLimit = miscRateLimit.Resources.Core.Limit;
+            // Get the Minutes till reset.
+            TimeSpan span = miscRateLimit.Resources.Core.Reset.UtcDateTime.Subtract(DateTime.UtcNow);
+            // In the message, cast TotalMinutes to an int to get a whole number of minutes.
+            string rateLimitMessage = $"Limit={miscRateLimit.Resources.Core.Limit}, Remaining={miscRateLimit.Resources.Core.Remaining}, Limit Reset in {(int)span.TotalMinutes} minutes.";
+            if (prependMessage != null)
+            {
+                rateLimitMessage = $"{prependMessage} {rateLimitMessage}";
+            }
+            Console.WriteLine(rateLimitMessage);
+        }
+
+        // Given a teamId, get the Team from github. Chances are, this is only going to be used to get
+        // the first team
+        public async Task<Team> GetTeamById(int teamId)
+        {
+            return await _gitHubClient.Organization.Team.Get(teamId);
+        }
+
+        /// <summary>
+        /// Given an Octokit.Team, call to get the team members. Note: GitHub's GetTeamMembers API gets all of the Users
+        /// for the team which includes all the members of child teams.
+        /// </summary>
+        /// <param name="team">Octokit.Team, the team whose members to retrieve.</param>
+        /// <returns>IReadOnlyList of Octokit.Users</returns>
+        /// <exception cref="ApplicationException">Thrown if GetAllMembers fails after all retries have been exhausted.</exception>
+        public async Task<IReadOnlyList<User>> GetTeamMembers(Team team)
+        {
+            ApiOptions apiOptions = new ApiOptions();
+            apiOptions.PageSize = 100;
+
+            // For the cases where exceptions/retries fail and an empty ReadOnlyList needs to be returned
+            List<User> emptyUserList = new List<User>();
+
+            int tryNumber = 0;
+            while (tryNumber < _numRetries)
+            {
+                tryNumber++;
+                try
+                {
+                    return await _gitHubClient.Organization.Team.GetAllMembers(team.Id, apiOptions);
+                }
+                // This is what gets thrown if we try and get a userList for certain special teams on GitHub. 
+                // None of these teams are used directly in anything and neither team is a child team of 
+                // azure-sdk-write. If a ForbiddenException is encountered, then report it and return an
+                // empty list.
+                catch (Octokit.ForbiddenException forbiddenEx)
+                {
+                    Console.WriteLine($"{team.Name} cannot be retrieved using a GitHub PAT.");
```

Comment:
```suggestion
                    Console.WriteLine($"Team named {team.Name} cannot be retrieved using a GitHub PAT.");
```

Reply:
We're in a function that gets members for a team, which can throw the ForbiddenException if it can't be retrieved via a PAT. Furthermore, I'm outputting the exception whose message is "Exception trying to get members for Team: `Team Name`"